app/controllers/tags_controller.rb

```diff
@@ -0,0 +1,31 @@
+class TagsController < ApplicationController
+  before_action :set_tag, only: [:show, :edit, :update, :destroy]
+
+  def index
+    respond_to do |format|
+      format.html do
+        @tags = Tag.order('created_at DESC')
+      end
+
+      format.json do
+        render json: Tag.where('title LIKE ?', params[:term] + "%").map(&:title)
+      end
+    end
+  end
+
+  def show
+    @tag = Tag.find(params[:id])
```

Comment:
You can remove this line because you are setting the `@tag` using your `before_action`